### PR TITLE
Remove Newspack Dependency

### PR DESCRIPTION
**Pull request by justinshreve**
This PR updates the onboarding code to use `@wordpress/components` equivalents for `Button` and `CheckboxGroup`, cleans up our build code, and implements a new `ImageUpload` component. 

Since we previously switched to our own TextControl and SelectControl components, we were using  Newspack components for pretty basic things like checkboxes and buttons, both of which were just CSS changes. The largest blocker here was an ImageUpload component, that I ended up implementing in our component's package.

### Screenshots

Screen Shot 2019-11-25 at 2 43 50 PM

Screen Shot 2019-11-25 at 2 43 43 PM

Screen Shot 2019-11-25 at 4 02 09 PM

Screen Shot 2019-11-25 at 4 02 33 PM

### Detailed test instructions:

* Test various sections of the app including analytics, analytics settings, etc.
* Enable the profile wizard and walk through the steps, verify things look OK around buttons and checkboxes.
* Test the appearance step of the task list, upload and remove an image.
* Optional: Look at `ImageUpload` in the dev docs.

### Changelog Note:

Dev: Remove Newspack dependency.

### CODE REVIEW

**Comment by jeffstieler** [comment]
Looks like some [SASS linter errors](https://travis-ci.org/woocommerce/woocommerce-admin/jobs/616885136?utm_medium=notification&utm_source=github_status) broke the build.

Can we bring the [unit test](https://github.com/Automattic/newspack-plugin/blob/master/assets/components/src/image-upload/imageUpload.test.js) over from the Newspack repo as well?

File: client/dashboard/style.scss [hunk 1]
```diff
@@ -188,3 +153,29 @@
 		}
 	}
 }
+
+.woocommerce-task-dashboard__container .components-button.is-button,
+.woocommerce-profile-wizard__body .components-button.is-button,
+.woocommerce-page .components-modal__frame .components-button.is-button {
```
Comment (joshuatf) [inline client/dashboard/style.scss]: In Firefox and Safari, it looks like the vertical alignment isn't working for buttons:

Screen Shot 2019-11-26 at 3 36 25 PM
Screen Shot 2019-11-26 at 3 35 50 PM

**Comment by justinshreve** [comment]
Thanks for the feedback and testing!

Linter errors have been fixed, and display issues in FF/Safari have been fixed.

Screen Shot 2019-11-26 at 11 23 20 AM

File: packages/components/src/image-upload/test/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * Internal dependencies
+ */
+import ImageUpload from '../index';
+
+describe( 'ImageUpload', () => {
```
Comment (joshuatf) [inline packages/components/src/image-upload/test/index.js]: Nice 👍

File: client/dashboard/task-list/style.scss [hunk 3]
```diff
@@ -182,7 +184,8 @@
 		position: relative;
 
 		.muriel-input-text input {
-			left: $gap * 2;
+			left: $gap;
+			max-width: 90%;
```
Comment (joshuatf) [inline client/dashboard/task-list/style.scss]: Minor nitpick: probably won't affect any screen sizes, but we can change to the following to fill this area snuggly:

```suggestion
			max-width: calc(100% - #{$gap});
```

**Pull request merged**